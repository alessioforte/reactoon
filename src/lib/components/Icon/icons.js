export default  {
    add: {
        d: 'M492,236H276V20c0-11.046-8.954-20-20-20c-11.046,0-20,8.954-20,20v216H20c-11.046,0-20,8.954-20,20s8.954,20,20,20h216v216c0,11.046,8.954,20,20,20s20-8.954,20-20V276h216c11.046,0,20-8.954,20-20C512,244.954,503.046,236,492,236z',
        width: '512'
    },
    caret: {
        d: 'M405.4,493.6l-387-386.2c-24.6-24.6-24.6-64.4,0-89c24.6-24.6,64.5-24.6,89.1,0L450,360.2L792.4,18.4c24.6-24.6,64.5-24.6,89.1,0c24.6,24.6,24.6,64.4,0,89l-387,386.2C482.2,505.9,466.1,512,450,512C433.9,512,417.7,505.9,405.4,493.6z',
        width: '900'
    },
    checkmark: {
        d: 'M71.27,133.67a8.75,8.75,0,0,1-6,2.73,7.68,7.68,0,0,1-3.49-.83,11.51,11.51,0,0,1-2.9-2.22C57,131.43,15.78,90.26,15.1,89.2A8.53,8.53,0,0,1,13.84,86,8.1,8.1,0,0,1,16,79.38c0.9-1,1.9-2,2.86-2.91l5.93-5.93c0.9-.89,1.76-1.82,2.71-2.66a8.17,8.17,0,0,1,6.64-2.2,8.43,8.43,0,0,1,3.12,1.22c1,0.65,27.73,27.59,27.92,27.79L128.31,31.5c0.74-.74,1.46-1.51,2.24-2.22a8.69,8.69,0,0,1,5.11-2.36,8,8,0,0,1,5.29,1.63c0.76,0.55,12.4,12.2,13,13a8,8,0,0,1,1.54,5.33,8.71,8.71,0,0,1-2.33,4.94c-0.68.76-1.44,1.46-2.16,2.19l-2.28,2.28-77.4,77.4h0Z',
        width: '512'
    },
    delete: {
        d: 'M494.7,100.8L339.6,256l155.1,155.1c23.1,23.1,23.1,60.5,0,83.6C483.2,506.2,468,512,452.9,512c-15.1,0-30.3-5.8-41.8-17.3L256,339.5L100.8,494.7C89.3,506.2,74.2,512,59.1,512c-15.1,0-30.2-5.8-41.8-17.3c-23.1-23.1-23.1-60.5,0-83.6L172.4,256L17.3,100.8c-23.1-23.1-23.1-60.5,0-83.6c23.1-23.1,60.5-23.1,83.5,0L256,172.4L411.1,17.3c23.1-23.1,60.5-23.1,83.5,0C517.8,40.4,517.8,77.8,494.7,100.8z',
        width: '512'
    },
    search: {
        d: 'M495,466.2L377.2,348.4c29.2-35.6,46.8-81.2,46.8-130.9C424,103.5,331.5,11,217.5,11C103.4,11,11,103.5,11,217.5   S103.4,424,217.5,424c49.7,0,95.2-17.5,130.8-46.7L466.1,495c8,8,20.9,8,28.9,0C503,487.1,503,474.1,495,466.2z M217.5,382.9   C126.2,382.9,52,308.7,52,217.5S126.2,52,217.5,52C308.7,52,383,126.3,383,217.5S308.7,382.9,217.5,382.9z',
        width: '512'
    },
    default: {
        d: '',
        width: '512'
    }
}