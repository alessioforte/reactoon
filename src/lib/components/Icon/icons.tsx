import React from 'react'

const icons: any = {
  add: {
    d:
      'M492,236H276V20c0-11.046-8.954-20-20-20c-11.046,0-20,8.954-20,20v216H20c-11.046,0-20,8.954-20,20s8.954,20,20,20h216v216c0,11.046,8.954,20,20,20s20-8.954,20-20V276h216c11.046,0,20-8.954,20-20C512,244.954,503.046,236,492,236z',
    width: '512'
  },
  caret: {
    d:
      'M405.4,493.6l-387-386.2c-24.6-24.6-24.6-64.4,0-89c24.6-24.6,64.5-24.6,89.1,0L450,360.2L792.4,18.4c24.6-24.6,64.5-24.6,89.1,0c24.6,24.6,24.6,64.4,0,89l-387,386.2C482.2,505.9,466.1,512,450,512C433.9,512,417.7,505.9,405.4,493.6z',
    width: '900'
  },
  checkmark: {
    d:
      'M256.2,512c-14.4,0-28.9-5.5-39.9-16.5L16.8,295.9c-22-22-22-57.8,0-79.8s57.8-22,79.8,0l159.6,159.6L615.4,16.5c22-22,57.8-22,79.8,0c22.1,22,22.1,57.8,0,79.8L296.1,495.5C285.1,506.5,270.7,512,256.2,512z',
    width: '712'
  },
  delete: {
    d:
      'M494.7,100.8L339.6,256l155.1,155.1c23.1,23.1,23.1,60.5,0,83.6C483.2,506.2,468,512,452.9,512c-15.1,0-30.3-5.8-41.8-17.3L256,339.5L100.8,494.7C89.3,506.2,74.2,512,59.1,512c-15.1,0-30.2-5.8-41.8-17.3c-23.1-23.1-23.1-60.5,0-83.6L172.4,256L17.3,100.8c-23.1-23.1-23.1-60.5,0-83.6c23.1-23.1,60.5-23.1,83.5,0L256,172.4L411.1,17.3c23.1-23.1,60.5-23.1,83.5,0C517.8,40.4,517.8,77.8,494.7,100.8z',
    width: '512'
  },
  search: {
    d:
      'M495,466.2L377.2,348.4c29.2-35.6,46.8-81.2,46.8-130.9C424,103.5,331.5,11,217.5,11C103.4,11,11,103.5,11,217.5   S103.4,424,217.5,424c49.7,0,95.2-17.5,130.8-46.7L466.1,495c8,8,20.9,8,28.9,0C503,487.1,503,474.1,495,466.2z M217.5,382.9   C126.2,382.9,52,308.7,52,217.5S126.2,52,217.5,52C308.7,52,383,126.3,383,217.5S308.7,382.9,217.5,382.9z',
    width: '512'
  },
  default: {
    d: '',
    width: '512',
    children: (
      <>
        <circle cx="151.3" cy="186.2" r="34.9"/>
        <circle cx="360.7" cy="186.2" r="34.9"/>
        <path d="M256,0C114.8,0,0,114.8,0,256s114.8,256,256,256s256-114.8,256-256S397.2,0,256,0z M151.3,256c-38.6,0-69.8-31.3-69.8-69.8
          s31.3-69.8,69.8-69.8s69.8,31.3,69.8,69.8S189.8,256,151.3,256z M360.7,256c-38.6,0-69.8-31.3-69.8-69.8s31.3-69.8,69.8-69.8
          s69.8,31.3,69.8,69.8S399.3,256,360.7,256z"/>
      </>
    )
  }
};

export const availableIcons = Object.keys(icons)

export default icons
