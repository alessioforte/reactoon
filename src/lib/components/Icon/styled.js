import styled from 'styled-components'

export const Svg = styled.svg`
    fill: ${props => props.color};
    width: ${props => props.width};
    margin: ${ props => props.margin || 0};
    transform: ${props => `rotate(${props.rotate || 0}deg)`};
`
export const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`
