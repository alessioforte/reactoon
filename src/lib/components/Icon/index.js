import React from 'react';
import PropTypes from 'prop-types';
import { Svg } from './styled';
import icons from './icons';

const Icon = ({ name, color, size }) => {
  if (!icons[name]) {
    console.error(`
    <Icon name='${name}' ... />
    '${name}' icon name is not defined in icons object try:
    ${Object.keys(icons).join(', ')}
    `);
  }
  const { d, width } = icons[name] || icons.default;
  return (
    <Svg
      color={color}
      height={size}
      x='0px'
      y='0px'
      viewBox={`0 0 ${width} 512`}
    >
      <path d={d} />
    </Svg>
  );
};

Icon.propTypes = {
  name: PropTypes.string.isRequired,
  color: PropTypes.string,
  size: PropTypes.string
};

Icon.defaultProps = {
  color: '#DADEE1',
  size: '30px'
};

export default Icon;
