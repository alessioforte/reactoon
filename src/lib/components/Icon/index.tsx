import React, { FC } from 'react';
import PropTypes from 'prop-types';
import { Svg } from './styled';
import icons from './icons';

const availableIcons = Object.keys(icons)

type Props = {
  name?: string,
  color?: string,
  size?: string,
}

const Icon: FC<Props> = ({ name = 'default', color, size }) => {
  const { d, width, children } = icons[name] || icons.default;
  return (
    <Svg
      color={color}
      height={size}
      x='0px'
      y='0px'
      viewBox={`0 0 ${width} 512`}
    >
      {children ? children : <path d={d} /> }
    </Svg>
  );
};

Icon.propTypes = {
  name: PropTypes.oneOf(availableIcons).isRequired,
  color: PropTypes.string,
  size: PropTypes.string
};

Icon.defaultProps = {
  name: 'default',
  color: '#DADEE1',
  size: '30px'
};

export default Icon;
