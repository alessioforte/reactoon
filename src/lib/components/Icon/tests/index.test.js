import React from 'react'
import {Â render } from '@testing-library/react'
import Icon from '../index'

describe('<Icon />', () => {
  it('should render and match the snapshot', () => {
    const { container: { firstChild } } = render(<Icon />)
    expect(firstChild).toMatchSnapshot()
  })
})
