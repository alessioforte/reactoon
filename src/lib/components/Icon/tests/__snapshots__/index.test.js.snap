// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Icon /> should render and match the snapshot 1`] = `
<svg
  class="sc-bdVaJa cdZYju"
  color="#DADEE1"
  height="30px"
  viewBox="0 0 512 512"
  x="0px"
  y="0px"
>
  <circle
    cx="151.3"
    cy="186.2"
    r="34.9"
  />
  <circle
    cx="360.7"
    cy="186.2"
    r="34.9"
  />
  <path
    d="M256,0C114.8,0,0,114.8,0,256s114.8,256,256,256s256-114.8,256-256S397.2,0,256,0z M151.3,256c-38.6,0-69.8-31.3-69.8-69.8 s31.3-69.8,69.8-69.8s69.8,31.3,69.8,69.8S189.8,256,151.3,256z M360.7,256c-38.6,0-69.8-31.3-69.8-69.8s31.3-69.8,69.8-69.8 s69.8,31.3,69.8,69.8S399.3,256,360.7,256z"
  />
</svg>
`;
